packages/entities/entities-shared/src/components/entity-empty-state/EntityEmptyState.vue

```diff
@@ -0,0 +1,230 @@
+        <span v-if="$slots['title-after']">
+          <slot name="title-after" />
+        </span>
+      </div>
+      <div
+        v-if="description || $slots.default"
+        class="entity-empty-state-description"
+      >
+        <slot name="default">
+          <p>
+            {{ description }}
+          </p>
+        </slot>
+      </div>
+      <div
+        v-if="pricing"
+        class="entity-empty-state-pricing"
+      >
+        <p>
+          <b>{{ t('emptyState.pricingTitle') }}</b> {{ pricing }}
+        </p>
+      </div>
+    </div>
+    <div
+      v-if="$slots.message"
+      class="entity-empty-state-message"
+    >
+      <slot name="message" />
+    </div>
+    <div class="entity-empty-state-action">
+      <KButton
+        v-if="(actionButtonText) || $slots.action"
+        appearance="primary"
+        size="large"
+        @click="$emit('create-button-clicked')"
+      >
+        <AddIcon />
+        {{ actionButtonText }}
+      </KButton>
+      <KButton
+        v-if="learnMoreLink"
+        appearance="secondary"
+        size="large"
+        @click="$emit('learning-hub-button-clicked')"
+      >
+        <BookIcon decorative />
+        {{ t('emptyState.learnMore') }}
+      </KButton>
+    </div>
+    <div class="entity-empty-state-card-container">
+      <template
+        v-for="feature in features"
+        :key="feature"
+      >
+        <KCard class="entity-empty-state-card">
+          <template #title>
+            <component
+              :is="getEntityIcon(feature.iconVariant)"
+              :color="`var(--kui-color-text-neutral-stronger, ${KUI_COLOR_TEXT_NEUTRAL_STRONGER})`"
+              :size="KUI_ICON_SIZE_40"
+            />
+            <div class="card-title">
+              {{ feature.title }}
+            </div>
+          </template>
+          {{ feature.description }}
+        </KCard>
+      </template>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { type PropType } from 'vue'
+import { KButton } from '@kong/kongponents'
+import { BookIcon, AddIcon, DeployIcon, PlugIcon, ChartDataIcon, AnalyticsIcon } from '@kong/icons'
+import composables from '../../composables'
+import type { EmptyStateFeature } from 'src/types/entity-empty-state'
+import { KUI_ICON_SIZE_40, KUI_COLOR_TEXT_NEUTRAL_STRONGER } from '@kong/design-tokens'
+
+const getEntityIcon = (iconType: string): object => {
+  console.log(iconType)
+  switch (iconType) {
+    case 'deploy':
+      return DeployIcon
+    case 'plug':
+      return PlugIcon
+    case 'chartData':
+      return ChartDataIcon
+    case 'analytics':
+      return AnalyticsIcon
+    default:
+      return BookIcon
+  }
+}
+
+defineProps({
+  title: {
+    type: String,
+    required: true,
+  },
+  description: {
+    type: String,
+    required: true,
+  },
+  pricing: {
+    type: String,
+    default: '',
+  },
+  actionButtonText: {
+    type: String,
+    default: '',
+  },
+  learnMoreLink: {
+    type: String,
```

Comment:
Should be `boolean` since we are emiting now.

Reply:
Updated